### PR TITLE
Fix coredns wait when there are no nodes

### LINKED ISSUE #5570: do not wait for coredns if there are no nodegroups is broken
### What were you trying to accomplish?

Create a cluster without nodegroups, this was recently fixed in https://github.com/weaveworks/eksctl/pull/5458

### What happened?

In https://github.com/weaveworks/eksctl/pull/5458 the change is tested by @Skarlso with:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: gb-test-ipv6-addon-01
  region: 'us-west-2'
  version: '1.22'

addons:
  - name: vpc-cni
  - name: coredns
  - name: kube-proxy
iam:
  withOIDC: true


kubernetesNetworkConfig:
  ipFamily: IPv6
```

But, after this test @cPu1 requested changes https://github.com/weaveworks/eksctl/pull/5458#discussion_r904736071, where:

```
if len(pool) == 0 && addon.Name == api.CoreDNSAddon {
```

is changed to
```
if addon.Name == api.CoreDNSAddon && (len(a.clusterConfig.NodeGroups) > 0 || len(a.clusterConfig.ManagedNodeGroups) > 0) {
```
eg it requires nodegroups when it shouldnt.

Shouldn't this code be:

```
if addon.Name == api.CoreDNSAddon && (len(a.clusterConfig.NodeGroups) == 0 && len(a.clusterConfig.ManagedNodeGroups) == 0) {
```

?

### How to reproduce it?

Using latest eksctl with the config above:

```
$ eksctl create cluster -f skariso.yml
2022-08-03 15:34:21 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-cluster"
2022-08-03 15:35:22 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-cluster"
2022-08-03 15:37:31 [ℹ]  daemonset "kube-system/aws-node" restarted
2022-08-03 15:37:31 [ℹ]  creating role using recommended policies
2022-08-03 15:37:32 [ℹ]  deploying stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 15:37:32 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 15:38:03 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 15:39:02 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 15:40:26 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 15:40:27 [ℹ]  creating addon
2022-08-03 15:40:39 [ℹ]  addon "vpc-cni" active
2022-08-03 15:40:39 [ℹ]  waiting for the control plane availability...
2022-08-03 15:40:44 [✔]  saved kubeconfig as "/root/.kube/config"
2022-08-03 15:40:44 [ℹ]  no tasks
2022-08-03 15:40:44 [✔]  all EKS cluster resources for "gb-test-ipv6-addon-01" have been created
2022-08-03 15:40:45 [ℹ]  no recommended policies found, proceeding without any IAM
2022-08-03 15:40:45 [ℹ]  creating addon
2022-08-03 15:40:57 [!]  1 error(s) occurred while creating addons
2022-08-03 15:40:57 [✖]  addon status transitioned to "DEGRADED"
Error: failed to create addons
```

### PR DESCRIPTION
**Pull request by matti**
### Description

Fixes #5570, tested manually with

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: gb-test-ipv6-addon-01
  region: 'us-west-2'
  version: '1.22'

addons:
  - name: vpc-cni
  - name: coredns
  - name: kube-proxy
iam:
  withOIDC: true


kubernetesNetworkConfig:
  ipFamily: IPv6
```

```
...
2022-08-03 16:32:52 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-cluster"
2022-08-03 16:33:53 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-cluster"
2022-08-03 16:36:02 [ℹ]  daemonset "kube-system/aws-node" restarted
2022-08-03 16:36:02 [ℹ]  creating role using recommended policies
2022-08-03 16:36:03 [ℹ]  deploying stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 16:36:03 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 16:36:34 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 16:37:10 [ℹ]  waiting for CloudFormation stack "eksctl-gb-test-ipv6-addon-01-addon-vpc-cni"
2022-08-03 16:37:12 [ℹ]  creating addon
2022-08-03 16:37:23 [ℹ]  addon "vpc-cni" active
2022-08-03 16:37:23 [ℹ]  waiting for the control plane availability...
2022-08-03 16:37:28 [✔]  saved kubeconfig as "/root/.kube/config"
2022-08-03 16:37:28 [ℹ]  no tasks
2022-08-03 16:37:28 [✔]  all EKS cluster resources for "gb-test-ipv6-addon-01" have been created
2022-08-03 16:37:29 [ℹ]  no recommended policies found, proceeding without any IAM
2022-08-03 16:37:29 [ℹ]  creating addon
2022-08-03 16:37:29 [ℹ]  no recommended policies found, proceeding without any IAM
2022-08-03 16:37:29 [ℹ]  creating addon
2022-08-03 16:37:40 [ℹ]  addon "kube-proxy" active
2022-08-03 16:37:53 [ℹ]  kubectl command should work with "/root/.kube/config", try 'kubectl get nodes'
2022-08-03 16:37:53 [✔]  EKS cluster "gb-test-ipv6-addon-01" in "us-west-2" region is ready
```

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/actions/addon/addon.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func New(clusterConfig *api.ClusterConfig, eksAPI awsapi.EKS, stackManager manag
 func (a *Manager) waitForAddonToBeActive(ctx context.Context, addon *api.Addon, waitTimeout time.Duration) error {
 	// We don't wait for coredns if there are no nodegroups. It will get into degraded state
 	// and recover once nodegroups are added.
-	if addon.Name == api.CoreDNSAddon && (len(a.clusterConfig.NodeGroups) > 0 || len(a.clusterConfig.ManagedNodeGroups) > 0) {
+	if addon.Name == api.CoreDNSAddon && (len(a.clusterConfig.NodeGroups) == 0 && len(a.clusterConfig.ManagedNodeGroups) == 0) {
```
Comment (cPu1) [inline pkg/actions/addon/addon.go:44]: As noted in the linked [issue](https://github.com/weaveworks/eksctl/issues/5570#issuecomment-1204207831), the condition should be

```suggestion
	if addon.Name == api.CoreDNSAddon && !a.clusterConfig.HasNodes() {
```

**Review by cPu1 — CHANGES_REQUESTED** [review]
Thanks for the PR. I have left a comment to change the condition to account for nodegroups with zero nodes.

**Comment by matti** [comment]
@cPu1 changed to that now.

**Review by cPu1 — APPROVED** [review]
Thanks for the quick fix!

**Review by Himangini — APPROVED** [review]

**Pull request merged**